internal/helpers/kube/kube.go

```diff
@@ -210,3 +211,45 @@ func (kh *KubeHelper) BuildOwnerReference() metav1.OwnerReference {
 	}
 	return ownership
 }
+
+const (
+	SECRET    = "Secret"
+	CONFIGMAP = "ConfigMap"
+)
+
+func (kh *KubeHelper) GetValueFrom(ctx context.Context, kind, namespace, name, key string) (string, error) {
+	nsName := types.NamespacedName{
+		Namespace: namespace,
+		Name:      name,
+	}
+	switch kind {
+	case SECRET:
+		obj := &corev1.Secret{}
+		if err := kh.Cli.Get(ctx, nsName, obj); err != nil {
+			return "", err
+		}
+		val, ok := GetValueByKey(obj.Data, key)
+		if !ok {
+			return "", fmt.Errorf("secret %s doesn't contain key %s", name, key)
+		}
+		return string(val), nil
+	case CONFIGMAP:
+		obj := &corev1.ConfigMap{}
+		if err := kh.Cli.Get(ctx, nsName, obj); err != nil {
+			return "", err
+		}
+		val, ok := GetValueByKey(obj.Data, key)
+		if !ok {
+			return "", fmt.Errorf("secret %s doesn't contain key %s", name, key)
+		}
+		return val, nil
+	default:
+		err := fmt.Errorf("unknown source kine: %s", kind)
```

Comment:
Maybe more `Unsupported source kind %s, please use "configmap" or "secret"`?

Reply:
Should look better now